crates/types/src/traits/block_contents.rs

```diff
@@ -42,24 +50,31 @@ pub trait BlockPayload:
     /// The type of the transitions we are applying
     type Transaction: Transaction;
 
+    // type Header: BlockHeader;
+
     /// returns hashes of all the transactions in this block
     /// TODO make this ordered with a vec
     fn contained_transactions(&self) -> HashSet<Commitment<Self::Transaction>>;
 }
 
-// TODO (Keyao) Determine whether we can refactor BlockPayload and Transaction from traits to structs.
-// <https://github.com/EspressoSystems/HotShot/issues/1815>
-/// Abstraction over any type of transaction. Used by [`BlockPayload`].
-pub trait Transaction:
-    Clone + Serialize + DeserializeOwned + Debug + PartialEq + Eq + Sync + Send + Committable + Hash
+/// Header of a block, which commits to a [`BlockPayload`].
+pub trait BlockHeader:
+    Serialize + Clone + Debug + Hash + PartialEq + Eq + Send + Sync + DeserializeOwned
 {
+    /// Block payload associated with the commitment.
+    type Payload: BlockPayload;
+
+    /// Get the payload commitment.
+    fn commitment(&self) -> Commitment<Self::Payload>;
```

Comment:
Same here, I think `payload_commitment` would be slightly better (otherwise if the header type is `Committable` it could get confusing that `header.commit()` and `header.commitment()` commit to two completely different objects)